server/sonar-server/src/test/java/org/sonar/server/permission/ws/UsersActionTest.java

```diff
@@ -0,0 +1,141 @@
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.permission.ws;
+
+import com.google.common.collect.Lists;
+import java.util.Collections;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.ArgumentCaptor;
+import org.sonar.api.server.ws.WebService.Param;
+import org.sonar.api.server.ws.WebService.SelectionMode;
+import org.sonar.api.utils.System2;
+import org.sonar.core.permission.GlobalPermissions;
+import org.sonar.core.permission.UserWithPermission;
+import org.sonar.db.DbTester;
+import org.sonar.db.permission.PermissionQuery;
+import org.sonar.db.user.GroupMembershipQuery;
+import org.sonar.server.permission.PermissionService;
+import org.sonar.server.permission.UserWithPermissionQueryResult;
+import org.sonar.server.ws.TestResponse;
+import org.sonar.server.ws.WsActionTester;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.sonar.test.JsonAssert.assertJson;
+
+public class UsersActionTest {
+
+  @Rule
+  public DbTester db = DbTester.create(System2.INSTANCE);
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+  WsActionTester ws;
+  PermissionService permissionService;
+
+  UsersAction underTest;
+
+  @Before
+  public void setUp() {
+    permissionService = mock(PermissionService.class);
+    underTest = new UsersAction(permissionService);
+    ws = new WsActionTester(underTest);
+  }
+
+  @Test
+  public void search_for_users_with_one_permission() {
+    List<UserWithPermission> users = Lists.newArrayList(
+      new UserWithPermission()
+        .setLogin("first-login")
+        .setName("first-name")
+        .hasPermission(true),
+      new UserWithPermission()
+        .setLogin("second-login")
+        .setName("second-name")
+        .hasPermission(true)
+      );
+    ArgumentCaptor<PermissionQuery> permissionQueryCaptor = ArgumentCaptor.forClass(PermissionQuery.class);
+    when(permissionService.findUsersWithPermission(permissionQueryCaptor.capture())).thenReturn(new UserWithPermissionQueryResult(users, 2));
+
+    TestResponse result = ws.newRequest().setParam("permission", "scan").execute();
+
+    assertJson(result.getInput()).isSimilarTo(ws.getTestFile(getClass(), "users.json"));
```

Comment:
Yes, you should, and you can replace what you've find elsewhere by this.
It makes no sense to re-invent the wheel !